modules/core/24-host/validate.go

```diff
@@ -25,7 +25,7 @@ var DefaultMaxPortCharacterLength = 128
 // - Alphanumeric
 // - `.`, `_`, `+`, `-`, `#`
 // - `[`, `]`, `<`, `>`
-var IsValidID = regexp.MustCompile(`^[a-zA-Z0-9\.\_\+\-\#\[\]\<\>]+$`).MatchString
+var IsValidID = regexp.MustCompile(`^[a-zA-Z0-9\.\_\+\-\#\|\[\]\<\>]+$`).MatchString
```

Comment:
I've added `|` to the valid characters in this regex. I intentionally left this in to raise on the PR. 
Figured as we are using `|` as the delimiter for the version string then it would be appropriate to use it for the controller port ID. For example: `ics27-1|0|0|cosmos17dtl0mjt3t77kpuhg2edqzjpszulwhgzuj9ljs`
This regex is used when ICA attempts to call `BindPort`

This allows cleaner splitting for connection id validation. If there is no issue with adding the pipe char to this regex let me know and I will open a PR with this one liner targeted for main!

Reply:
Hmm, this requires a ICS change. We should open a pr there first for discussion. I'm not sure if it was intentionally left out. Otherwise we should just pick a different character

Reply:
Cool, I'm happy to open a PR for discussion, but I'm open to just using a different character if you think that would be less hassle. But in which case I'd also suggest we change the version string delimiter to align for consistency. 

Aside from a completely hyphenated string which would make parsing awkward, I think dot separation looks best. PortID: `ics27-1.0.0.cosmos17dtl0mjt3t77kpuhg2edqzjpszulwhgzuj9ljs`
Version: `ics27-1.cosmos17dtl0mjt3t77kpuhg2edqzjpszulwhgzuj9ljs`

wdyt? Also tagging @seantking

Reply:
Dot separation is fine with me, although I prefer the |

Reply:
Dot separation looks good to me. ICA spec needs to be updated anyways, so lets open a pr there and point out both options. For now, I'd prefer to go with dot separation so we don't have to modify this code

Reply:
I can open a PR over there, I've reverted the change to regex in `24-host` and updated the delimiter to `"."`